### PR TITLE
Add a old settings back and move new ones to advanced settings

### PR DESCRIPTION
**Pull request by red-001**
Add back the old setting as the new ones can be very confusing, especially to new players. The new settings are now listed under the advanced settings tab.

### CODE REVIEW

**Comment by MoNTE48** [comment]
+1 :)

**Comment by Fixer-007** [comment]
+1
But too much tabs, maybe button with "advanced settings" inside the usual "Settings" tab?

**Comment by est31** [comment]
Perhaps there can be a check box "advanced settings" in the settings tab. if its checked, the advanced settings tab appears. if not, there is only the normal settings tab.

This PR once merged will fix #3486.

**Comment by est31** [comment]
can you fix these whitespace errors:
```
Wende an: Add a old settings back and move new ones to advanced settings
minetest/.git/rebase-apply/patch:318: space before tab in indent.
                                gui_scale_to_scrollbar() .. "]" ..
minetest/.git/rebase-apply/patch:370: space before tab in indent.
                                viewing_range_min_to_scrollbar() .. "]" ..
minetest/.git/rebase-apply/patch:377: space before tab in indent.
                                viewing_range_max_to_scrollbar() .. "]" ..
minetest/.git/rebase-apply/patch:378: space before tab in indent.
                        "tooltip[sb_viewing_range_max;" .. fgettext("Maximal Viewing Range: $1 nodes",
minetest/.git/rebase-apply/patch:391: space before tab in indent.
                                viewing_range_min_to_scrollbar() .. "]" ..
warning: 5 Zeilen fügen Whitespace-Fehler hinzu.
```

**Comment by est31** [comment]
Or the check box can toogle the settings tab with the advanced settings tab, only displaying one at a time. For the advanced settings tab, it can be a setting like every other.

**Comment by kilbith** [comment]
:-1: for adding a new tab (poor implementation). A button would be better in the current one.

Like @BlockMen did in his fork : https://github.com/blockplanet/blockplanet/commit/56d545b2a1f55bca827673e6537cd28499282a50

**Comment by est31** [comment]
@kilbith please stop linking code from GPL licensed forks of minetest, one shouldn't look at them.

**Comment by kilbith** [comment]
@est31 If there's an unique way of implementing something, I can't see how could we avoid it for licensing issues, especially if the code has no complex logic. Or maybe it was irony ?

**Comment by est31** [comment]
idk, its no good habit that's for sure.

**Comment by rubenwardy** [comment]
This adds a toggle to show the tab. Really simple and advanced should be in the same tab, but you toggle between them.

**Comment by kilbith** [comment]
Can be closed. https://github.com/minetest/minetest/pull/3514

File: builtin/mainmenu/init.lua [hunk 1]
```diff
@@ -120,7 +121,15 @@ local function init_globals()
 	end
 
 	-- Create main tabview
-	local tv_main = tabview_create("maintab",{x=12,y=5.2},{x=0,y=0})
+
```
Comment (est31) [inline builtin/mainmenu/init.lua]: whitespace error

File: builtin/mainmenu/init.lua [hunk 2]
```diff
@@ -131,7 +140,10 @@ local function init_globals()
 	else
 		tv_main:add(tab_simple_main)
 	end
-	tv_main:add(tab_settings)
+	tv_main:add(tab_simple_settings)
+	if core.setting_getbool("show_advanced_menu") then
```
Comment (est31) [inline builtin/mainmenu/init.lua]: Can the setting be called "show_advanced_settings_menu" ? Otherwise it might lead to confusion.

**Comment by est31** [comment]
:+1: if squashed and things I pointed out above are fixed. Probably there are some white space errors hiding, so merging person should check for them (I can't as it isn't squashed yet).

File: minetest.conf.example [hunk 3]
```diff
@@ -607,6 +607,9 @@
 #    type: bool
 # gui_scaling_filter = false
 
+#	When set to true will add a tab with advanced settings to main menu.
```
Comment (kilbith) [inline minetest.conf.example]: Don't use a tab here. Keep this text aligned with whitespaces.

File: builtin/mainmenu/tab_simple_settings.lua [hunk 4]
```diff
@@ -0,0 +1,530 @@
...
+				dump(core.setting_getbool("viewing_range_auto")) .. "]"..
+			"label[3.85,4.5;" .. fgettext("Max.") .. "]" ..
+			"scrollbar[4.65,4.55;2.75,0.4;sb_viewing_range_max;horizontal;" ..
+				viewing_range_max_to_scrollbar() .. "]" ..
+			"tooltip[sb_viewing_range_max;" .. fgettext("Maximal Viewing Range: $1 nodes",
+				get_vrange_max) .. "]"
+
+		if (get_vrange_min == nil and get_vrange_min > get_vrange_max) then
+			core.setting_set("viewing_range_nodes_min", get_vrange_max)
+		end
```
Comment (kilbith) [inline builtin/mainmenu/tab_simple_settings.lua]: Delete this from `if` to `end`, and add the following code just before `return tab_string` (line 357) : 

``` Lua
    if get_vrange_min > get_vrange_max then
        core.setting_set("viewing_range_nodes_min", get_vrange_max)
    end
```

It fixes a small bug when vrange min exceeds max (I coded this feature).

File: builtin/mainmenu/tab_simple_settings.lua [hunk 5]
```diff
@@ -0,0 +1,530 @@
...
+	elseif (core.setting_getbool("viewing_range_auto") == nil or
+			core.setting_getbool("viewing_range_auto") == false) then
```
Comment (kilbith) [inline builtin/mainmenu/tab_simple_settings.lua]: Replace `elseif ...` by `else`

**Comment by red-001** [comment]
@est31 squashed

**Comment by est31** [comment]
There is still the trailing whitespace I've pointed out above, otherwise okay.

**Comment by paramat** [comment]
Hmm i'm still not sure about this, it's good to have everything done in one place, and players will soon learn where each setting is if they are forced to use the new settings menu, although i admit it takes longer to open up the submenus.
Perhaps instead the new settings menu could display the most frequently used settings in a way that is quick to access, at menu top.
As much as i'd like to make people happy by giving a +1 i'm hesitant. the new settings menu was intended to replace after all, maybe it's best to focus on improving that instead.

**Comment by red-001** [comment]
@paramat Maybe it would be better if instead of two tabs the button opened the advanced settings in the same tab? It that's one click to enable shaders in the old menu and it that's 5 to do that very basic thing in the new one settings menu. As for making most often used settings easier to use that is exactly what my pull request does.

**Comment by paramat** [comment]
Yes i agree the branching structure of the new settings menu needs improvement to have the often-used settings just 1 or 2 clicks away.

**Comment by est31** [comment]
I don't think its messy, its unix way: more than one method to do sth.

And I also think it would be better if the simple settings tab would be hidden if the new settings tab is displayed. Then there is only one settings tab at a time.

**Comment by Gael-de-Sailly** [comment]
> […] and players will soon learn where each setting is if they are forced to use the new settings menu […]

If they are forced to use the new settings menu, they will quickly be bored, and maybe leave Minetest. The players are not all geeks, most of them simply want a game that works and that is easy to use. I've seen many players complaining on online servers, and on the forum. They can't get used to.
I've even seen some criticisms about it on dev IRC.

Please, merge this PR or another similar one before the next stable version. We need it. You, devs, maybe no, but the Minetest community in general, needs simpler settings.

New, little experienced players would have this simple settings tab with visually explicit checkboxes and dropdowns, and be happy. Geeks would only have to click one button to see they preferred interface, and be happy. We could even keep the settings tab status in memory, and show directly the advanced settings on the next time. All is well, isn't?

I know that I was the first to ask for this new settings table. I regret to have badly explained what I would have liked. I never wanted to remove the old settings tab, just having both, with a button to switch between the two interfaces, for example "Show / Hide advanced settings".

**Comment by paramat** [comment]
@red-001 see https://github.com/minetest/minetest/pull/3514#issuecomment-185633832 and following comments, perhaps you could see if you can improve this PR by looking at that code, as that PR will not be rebased.

**Comment by kilbith** [comment]
No, better a trustable contributor or dev to revive BlockMen's PR on his own. red-001 is none of this.

**Comment by paramat** [comment]
red-001 we're going ahead with a rebase of BlockMen's commit, so no need to work on it.

**Pull request closed without merging**